### PR TITLE
Update profiles to support both legacy XR and XR SDK

### PR DESCRIPTION
**Pull request by keveleigh**
## Overview

Created with @RogPodge!

Updates all profiles to have compatible data providers and controller visualization & mapping across both XR SDK and legacy XR. Adapts across Unity 2018, 2019, and 2020+ to spin up the correct providers in the correct contexts. Also deprecates all the existing XRSDK- and OpenXR-specific profiles, which are no longer needed.

## Changes
- Fixes: https://github.com/microsoft/MixedRealityToolkit-Unity/issues/9419, fixes https://github.com/microsoft/MixedRealityToolkit-Unity/issues/9378

### CODE REVIEW

**Review by RogPodge — APPROVED** [review]
The changes look good to me, I need to make some adjustments to this PR concerning the new null profile settings: https://github.com/MicrosoftDocs/mixed-reality-pr/pull/2926.

Another quick pass by someone who hasn't touched this feature heavily would be appreciated before merging.

File: Assets/MRTK/Core/Services/BaseDataProviderAccessCoreSystem.cs
Comment (MaxWang-MS) [inline Assets/MRTK/Core/Services/BaseDataProviderAccessCoreSystem.cs:219]: I guess this is a general comment rather than specific for this line, but I am wondering this changes in this PR can properly support scenarios where neither of the two pipelines are in use (e.g. maybe neither is used in the web verion of MRTK)?

**Comment by RogPodge** [comment]
image

image

Adding images to be included in documentation showing that profiles with missing types are now hidden from the user by default

**Comment by RogPodge** [comment]
image

image

Updated images again with better wording

**Review by davidkline-ms — APPROVED** [review]

File: Assets/MRTK/Core/Utilities/XRSettingsUtilities.cs
Comment (keveleigh) [inline Assets/MRTK/Core/Utilities/XRSettingsUtilities.cs:35]: @davidkline-ms @MaxWang-MS @RogPodge @CDiaz-MS I've found that the AppData folder on-device doesn't get cleaned up on an uninstall, so sometimes the subsystem lists are cached long term. For example from the logs (even though I'm running legacy XR on Unity 2019):

```
[Subsystems] Discovering subsystems at path C:/Data/Users/DefaultAccount/AppData/Local/DevelopmentFiles/Microsoft.MixedReality.ToolkitVS.Master_ARM64.kurti/Data/UnitySubsystems
[Subsystems] No descriptors matched for  examples in UnitySubsystems/MicrosoftOpenXRPlugin/UnitySubsystemsManifest.json.
[Subsystems] 1 'inputs' descriptors matched in UnitySubsystems/MicrosoftOpenXRPlugin/UnitySubsystemsManifest.json
[Subsystems] No descriptors matched for  displays in UnitySubsystems/MicrosoftOpenXRPlugin/UnitySubsystemsManifest.json.
[Subsystems] 1 'meshings' descriptors matched in UnitySubsystems/MicrosoftOpenXRPlugin/UnitySubsystemsManifest.json
[Subsystems] No descriptors matched for  examples in UnitySubsystems/UnityOpenXR/UnitySubsystemsManifest.json.
[Subsystems] 1 'inputs' descriptors matched in UnitySubsystems/UnityOpenXR/UnitySubsystemsManifest.json
[Subsystems] 1 'displays' descriptors matched in UnitySubsystems/UnityOpenXR/UnitySubsystemsManifest.json
[Subsystems] No descriptors matched for  meshings in UnitySubsystems/UnityOpenXR/UnitySubsystemsManifest.json.
[Subsystems] No descriptors matched for  examples in UnitySubsystems/WindowsMRXRSDK/UnitySubsystemsManifest.json.
[Subsystems] 1 'inputs' descriptors matched in UnitySubsystems/WindowsMRXRSDK/UnitySubsystemsManifest.json
[Subsystems] 1 'displays' descriptors matched in UnitySubsystems/WindowsMRXRSDK/UnitySubsystemsManifest.json
[Subsystems] 1 'meshings' descriptors matched in UnitySubsystems/WindowsMRXRSDK/UnitySubsystemsManifest.json
```

The only way I've found to properly clear the AppData folder is to set "Remove Non-Layout Files from Device" under the Remote Machine (not Device, unfortunately) Debugging settings in VS:

image

This will only affect people who installed a version of an app that supported XR SDK and has since rolled back to legacy (which I'd assume would be 100% developers only).

How do we want to approach this? Is this "edge case" enough, with a solid workaround, that we're okay with it? I'm not sure another way to check this info, and it works perfectly fine in-editor. It's just on-device.

Comment (RogPodge) [reply Assets/MRTK/Core/Utilities/XRSettingsUtilities.cs:35]: I think the use case of going from xrsdk to legacy should be uncommon enough that this shouldn't block the PR from getting in, though we might want to have a section in our documentation which explicitly lays out this step for developers who do so.

Comment (davidkline-ms) [reply Assets/MRTK/Core/Utilities/XRSettingsUtilities.cs:35]: I agree with @RogPodge. As long as we document this clearly, I think this PR should be fine.

The one thing I would be concerned with is the behavior for non-developer customers (ex: if they uninstall an XR SDK build of an app and attempt to install a legacy, older?, build).

Comment (davidkline-ms) [reply Assets/MRTK/Core/Utilities/XRSettingsUtilities.cs:35]: Also, do these messages break the app's behavior? Or just clutter the logs?

Comment (keveleigh) [reply Assets/MRTK/Core/Utilities/XRSettingsUtilities.cs:35]: Break the app's behavior. The app reports that it's running XR SDK (because stale subsystem json files are found), but Unity is actually running legacy. Thus, the providers are reading from the wrong APIs and no input comes through.

Comment (keveleigh) [reply Assets/MRTK/Core/Utilities/XRSettingsUtilities.cs:35]: > Thus, the providers are reading from the wrong APIs and no input comes through.

Actually, it goes a step further. The XR SDK providers see that their specific loader isn't running (because no XR SDK loader is running) and don't run themselves. So we end up with no truly-running provider!

File: Assets/MRTK/Core/Services/BaseDataProviderAccessCoreSystem.cs
Comment (keveleigh) [reply Assets/MRTK/Core/Services/BaseDataProviderAccessCoreSystem.cs]: I think currently, the default behavior won't be any different than what happens today. I agree that we might want to consider that case though! Happy to track that with a follow-up issue

**Pull request merged**
